NLog.Web.AspNetCore/LayoutRenderers/AspNetQueryStringLayoutRenderer.cs

```diff
@@ -0,0 +1,165 @@
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="logEvent"></param>
+        protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
+        {
+            if (this.QueryStringKeys?.Count > 0)
+            {
+                var httpRequest = HttpContextAccessor?.HttpContext?.TryGetRequest();
+
+                if (httpRequest == null)
+                    return;
+
+#if !NETSTANDARD_1plus
+                this.SerializeQueryString(builder, httpRequest.QueryString);
+#else
+                this.SerializeQueryString(builder, httpRequest.Query);
+#endif
+
+            }
+        }
+
+#if !NETSTANDARD_1plus
+        /// <summary>
+        /// To Serialize the QueryString based on the configuration.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="queryStrings"></param>
+        private void SerializeQueryString(StringBuilder builder, NameValueCollection queryStrings)
+        {
+            var includeArrayEndBraces = false;
+
+            if (queryStrings?.Count > 0)
+            {
+                var i = 0;
+                foreach (var configuredKey in this.QueryStringKeys)
+                {
+                    var value = queryStrings[configuredKey];
+
+                    if (!String.IsNullOrEmpty(value))
+                    {
+                        if (i > 0)
+                            builder.Append($",{Environment.NewLine}");
+
+                        switch (this.OutputFormat)
+                        {
+                            case AspNetLayoutOutputFormat.Flat:
+                                builder.Append($"{configuredKey}:{value}");
+                                break;
+                            case AspNetLayoutOutputFormat.Json:
+                                if (!includeArrayEndBraces)
+                                {
+                                    includeArrayEndBraces = true;
+                                    builder.Append(jsonArrayStartBraces);
+                                }
+                                builder.Append($"{jsonElementStartBraces}{doubleQuotes}{configuredKey}{doubleQuotes}:{doubleQuotes}{value}{doubleQuotes}{jsonElementEndBraces}");
+                                break;
+                            default:
+                                break;
+                        }
+                        i++;
+                    }
+                }
+
+                if (includeArrayEndBraces)
+                    builder.Append(jsonArrayEndBraces);
+            }
+        }
+#else
+        /// <summary>
+        /// To Serialize the QueryString based on the configuration.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="queryStrings"></param>
+        private void SerializeQueryString(StringBuilder builder, IQueryCollection queryStrings)
+        {
```

Comment:
Indeed, so you need transforming it to a common type. Only the transformation is platform specific.